File: 2289-102-2551-put-marbles-in-bags/2289-102-2551-put-marbles-in-bags.js
var putMarbles = function(weights, k) {
    const n = weights.length;
    const v = [];
    
    for (let i = 1; i < n; i++) {
        v.push(weights[i-1] + weights[i]);
    }
    
    v.sort((a, b) => a - b);
    let ans = 0;
    
    for (let i = 0; i < k-1; i++) {
        ans += v[n-2-i] - v[i];
    }
    
    return ans;
};

File: 2289-102-2551-put-marbles-in-bags/2289-102-2551-put-marbles-in-bags.py
class Solution:
    def putMarbles(self, weights: list[int], k: int) -> int:
        n = len(weights)
        v = []
        
        for i in range(1, n):
            v.append(weights[i-1] + weights[i])
        
        v.sort()
        ans = 0
        
        for i in range(k-1):
            ans += v[n-2-i] - v[i]
        
        return ans